### PR TITLE
4단계 - 멀티 모듈 적용

### PR DESCRIPTION
**Pull request by SHINMJ**
안녕하세요 리뷰어님!!!

메리 크리스마스입니다!!!! 크리스마스에도 코드 꼼꼼히 봐주셔서 너무 감사합니다!!!
크리스마스인데 저희 모두 열심히네요ㅠㅠ

멀티 모듈은 다음과 같이 나누었습니다.
- ui, application, domain 계층으로 나누고 
- domain에서 바운더리 컨텍스트로 나누었습니다.
- module-api (presentation layer)
- module-web (application layer)
- module-menu (domain layer, menu-product 도메인)
- module-order (domain layer, table-order 도메인)
- module-common (exception, vo 등 공통)

그럼 이번 미션도 잘 부탁드립니다!!!!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,4 +100,14 @@
 - [x] 메뉴정보가 변경되더라고 주문 항목이 변경되지 않도록 구현한다.
 - [x] 패키지, 클래스간 방향성 설정 - 되도록이면 단방향
 
-![의존관계.png](./uml/의존관계.png)
\ No newline at end of file
+![의존관계.png](./uml/의존관계.png)
```
Comment (jaeyeonling) [inline README.md:103]: 멀티모듈 설계 시 계층형와 도메인형을 같이 가져가도록 분리를 해주셨네요!
대부분 계층형을 선택하거나, 도메인형을 선택하는데, 두 가지 방법을 같이 사용하신 이유가 있을까요!?
두 방법을 같이 사용하다보니 각 계층형에 따른 장점이 조금 줄어드는 효과도 있는 것 같아 의견이 궁금합니다 ㅎㅎ

**Review by jaeyeonling — CHANGES_REQUESTED** [review]
멀티 모듈을 잘 적용해주셨습니다 👍 
소소한 질문 남겨두었으니 확인 부탁드릴게요~
메리 크리스마스입니다 🎅

File: README.md (hunk 1 above)
Comment (SHINMJ) [reply README.md:103]: 처음에는 도메인에 따라 분리를 할까 했었는데 메뉴와 상품은 긴밀하게 연결되어 보이고 테이블과 주문도 긴밀하게 연결되어 보여 menu-product, order-table 이렇게 바운더리 컨텍스트로 나누었습니다!! 거기에 공통 vo는 common으로 두었구요. 도메인형으로 나누다 보니 endpoint를 하나로 관리해야 겠다는 생각이 들어서 presentation 계층을 분리해봤습니다!!

api 영역을 분리해 내고 보니 서비스 레이어도 분리하는게 맞지 않나 라는 생각이 들어서요. 이부분은 처음 모듈 분리 시 제외했던 부분인데 잘못생각한 것 같기도 하고요..ㅋ  그래서 일단은 api -> web -> 각 도메인 이렇게 분리를 했었는데 지금와서 생각해보니 컨텍스트별로 나누는게 좋다고 생각합니다!!

그럼 최종적으로 
- module-api : api 어플리케이션 모듈
- module-common : 공통 exception , vo 가 있는 모듈 (타 모듈과 의존이 없는..)
- module-menu : 메뉴, 상품 도메인에 관한 모듈
- module-orderTable: 테이블, 주문 도메인에 관한 모듈

이 정도가 괜찮을 까요???  서비스 레이어는 각 도메인 모듈로 들어가는 것이 적당해 보이고요!!

Comment (SHINMJ) [reply README.md:103]: %3CmxGraphModel%3E%3Croot%3E%3CmxCell%20id%3D%220%22%2F%3E%3CmxCell%20id%3D%221%22%20parent%3D%220%22%2F%3E%3CmxCell%20id%3D%222%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BwhiteSpace%3Dwrap%3Bhtml%3D1%3BfontSize%3D14%3BfillColor%3D%23bac8d3%3BstrokeColor%3D%2323445d%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22260%22%20y%3D%22360%22%20width%3D%22490%22%20height%3D%22200%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%223%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BwhiteSpace%3Dwrap%3Bhtml%3D1%3BfontSize%3D14%3BfillColor%3D%23fad9d5%3BstrokeColor%3D%23ae4132%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22260%22%20y%3D%22220%22%20width%3D%22490%22%20height%3D%2290%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%224%22%20style%3D%22edgeStyle%3DorthogonalEdgeStyle%3Brounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BfontSize%3D14%3B%22%20edge%3D%221%22%20source%3D%225%22%20target%3D%228%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%225%22%20value%3D%22module-menu%22%20style%3D%22html%3D1%3BfontStyle%3D1%3BfontSize%3D14%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22290%22%20y%3D%22380%22%20width%3D%22140%22%20height%3D%2250%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%226%22%20style%3D%22edgeStyle%3DorthogonalEdgeStyle%3Brounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BfontSize%3D14%3BentryX%3D0.5%3BentryY%3D0%3BentryDx%3D0%3BentryDy%3D0%3B%22%20edge%3D%221%22%20source%3D%227%22%20target%3D%228%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%3E%3CmxPoint%20x%3D%22490%22%20y%3D%22490%22%20as%3D%22targetPoint%22%2F%3E%3CArray%20as%3D%22points%22%3E%3CmxPoint%20x%3D%22640%22%20y%3D%22460%22%2F%3E%3CmxPoint%20x%3D%22500%22%20y%3D%22460%22%2F%3E%3C%2FArray%3E%3C%2FmxGeometry%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%227%22%20value%3D%22module-orderTable%22%20style%3D%22html%3D1%3BfontStyle%3D1%3BfontSize%3D14%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22570%22%20y%3D%22380%22%20width%3D%22140%22%20height%3D%2250%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%228%22%20value%3D%22module-common%22%20style%3D%22html%3D1%3BfontStyle%3D1%3BfontSize%3D14%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22430%22%20y%3D%22490%22%20width%3D%22140%22%20height%3D%2250%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%229%22%20style%3D%22edgeStyle%3DorthogonalEdgeStyle%3Brounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BfontSize%3D14%3B%22%20edge%3D%221%22%20source%3D%2211%22%20target%3D%225%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2210%22%20style%3D%22edgeStyle%3DorthogonalEdgeStyle%3Brounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BfontSize%3D14%3B%22%20edge%3D%221%22%20source%3D%2211%22%20target%3D%227%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2211%22%20value%3D%22module-web%22%20style%3D%22html%3D1%3BfontStyle%3D1%3BfontSize%3D14%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22430%22%20y%3D%22240%22%20width%3D%22140%22%20height%3D%2250%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2212%22%20value%3D%22Application%20Layer%22%20style%3D%22text%3Bhtml%3D1%3BstrokeColor%3Dnone%3BfillColor%3Dnone%3Balign%3Dcenter%3BverticalAlign%3Dmiddle%3BwhiteSpace%3Dwrap%3Brounded%3D0%3BfontSize%3D14%3BfontStyle%3D1%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22280%22%20y%3D%22250%22%20width%3D%2260%22%20height%3D%2230%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2213%22%20value%3D%22Domain%26lt%3Bbr%26gt%3BLayer%22%20style%3D%22text%3Bhtml%3D1%3BstrokeColor%3Dnone%3BfillColor%3Dnone%3Balign%3Dcenter%3BverticalAlign%3Dmiddle%3BwhiteSpace%3Dwrap%3Brounded%3D0%3BfontSize%3D14%3BfontStyle%3D1%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22280%22%20y%3D%22490%22%20width%3D%2260%22%20height%3D%2230%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2214%22%20value%3D%22%22%20style%3D%22rounded%3D0%3BwhiteSpace%3Dwrap%3Bhtml%3D1%3BfontSize%3D14%3BfillColor%3D%23fff2cc%3BstrokeColor%3D%23d6b656%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22260%22%20y%3D%2290%22%20width%3D%22490%22%20height%3D%2290%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2215%22%20style%3D%22edgeStyle%3DorthogonalEdgeStyle%3Brounded%3D0%3BorthogonalLoop%3D1%3BjettySize%3Dauto%3Bhtml%3D1%3BexitX%3D0.5%3BexitY%3D1%3BexitDx%3D0%3BexitDy%3D0%3BentryX%3D0.5%3BentryY%3D0%3BentryDx%3D0%3BentryDy%3D0%3BfontSize%3D14%3B%22%20edge%3D%221%22%20source%3D%2216%22%20target%3D%2211%22%20parent%3D%221%22%3E%3CmxGeometry%20relative%3D%221%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2216%22%20value%3D%22module-api%22%20style%3D%22html%3D1%3BfontStyle%3D1%3BfontSize%3D14%3B%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22430%22%20y%3D%22110%22%20width%3D%22140%22%20height%3D%2250%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3CmxCell%20id%3D%2217%22%20value%3D%22Presentation%26lt%3Bbr%26gt%3B%26amp%3Bnbsp%3BLayer%22%20style%3D%22text%3Bhtml%3D1%3BstrokeColor%3Dnone%3BfillColor%3Dnone%3Balign%3Dcenter%3BverticalAlign%3Dmiddle%3BwhiteSpace%3Dwrap%3Brounded%3D0%3BfontSize%3D14%3BfontStyle%3D1%22%20vertex%3D%221%22%20parent%3D%221%22%3E%3CmxGeometry%20x%3D%22280%22%20y%3D%22120%22%20width%3D%2260%22%20height%3D%2230%22%20as%3D%22geometry%22%2F%3E%3C%2FmxCell%3E%3C%2Froot%3E%3C%2FmxGraphModel%3E

Comment (SHINMJ) [reply README.md:103]: 스크린샷 2021-12-26 오후 4 19 15

Comment (jaeyeonling) [reply README.md:103]: 설계에 정답은 없기 때문에 어떠한 의도로 설계 하였는지와 설계의 의도가 잘 담겨있는지가 중요할 것 같은데요!
처음 도메인으로 분리하는 것을 목적으로 하여 menu, order로 구분하신 점은 잘 해주신 것 같아요 👍 
하지만, presentation layer를 같이 사용한다면 결국 수정이 있을 때 같이 빌드를 해야하는 문제가 있을 것 같은데 어떻게 생각하시나요? ㅎㅎ

일반적으로 menu, order, common 과 같이 도메인 별로 레이어를 분리하는 경우와 presentation / application / domain과 같이 계층형으로 레이어를 구분할 경우의 핵심이 무엇인지 고민해보면 좋겠어요!!

Comment (SHINMJ) [reply README.md:103]: presentation layer 같은 경우에는 api만 존재하는데 각 모듈과 함께 따로 배포가 가능하고 API를 수정하지 않는 이상 도메인 레이어가 수정된다고 해서 같이 수정될 여지가 없는 것 같은데 api 모듈은 분리해도 되지 않을까요??

Comment (jaeyeonling) [reply README.md:103]: api 수정 없이 도메인의 수정이 있을 때 개별적으로 빌드가 되는 것이 셜계였다면 의도에 맞게 잘 설계가 된 것 같은데요!
menu와 관련된 api가 수정 됐을 때 order와 관련된 api도 같이 빌드되는 문제는 피할 수 없을 것 같아요 ㅎㅎ
위에도 말씀드렸지만 설계에 정답은 없기 때문에 의도한 내용이 잘 담겼는지가 더욱 중요할 것 같습니다!

Comment (SHINMJ) [reply README.md:103]: 그부분은 미처 생각을 못했습니다!!ㅠ 각 바우더리 컨텍스트별로 나눈다 라고 하면 menu-product, order-table 정도로 나누는게 적당한것 같아 다시 수정하였습니다!!

**Review by jaeyeonling — APPROVED** [review]
코멘트를 잘 반영해주셨습니다 👍 
미션 진행하시느라 정말 고생 많으셨어요!!
해당 요청은 머지하도록 할게요 마지막까지 화이팅입니다 😄

**Pull request merged**